src/FSharpLint.Client/LSPFSharpLintServiceTypes.fs

```diff
@@ -0,0 +1,70 @@
+﻿module FSharpLint.Client.LSPFSharpLintServiceTypes
+
+open System
+open System.Diagnostics
+open System.IO
+open StreamJsonRpc
+
+type FSharpLintResponseCode =
+    | ErrToolNotFound = -5
+    | ErrFileNotFound = -4
+    | ErrFilePathIsNotAbsolute = -3
+    | ErrCancellationWasRequested = -2
+    | ErrDaemonCreationFailed = -1
+    | OkCurrentDaemonVersion = 0
+
+type FSharpLintVersion = FSharpLintVersion of string
+type FSharpLintExecutableFile = FSharpLintExecutableFile of string
+type Folder = private Folder of string 
+with 
+    static member from (filePath: string) =
+        if File.Exists(filePath) then 
+            let folder = Path.GetFullPath(filePath) |> Path.GetDirectoryName
```

Comment:
if filePath is /foo/bar/baz, then folder var here is storing `baz` only, which is not what you want

Reply:
sorry, `bar` only, not `baz`, but what you should want to extract here is `/foo/bar`

Reply:
You made me doubt but `System.IO.Path.GetFullPath("/foo/bar/baz") |> System.IO.Path.GetDirectoryName` is indeed the expected `/foo/bar` :)

Reply:
Fair enough, but `(System.IO.Path.GetFullPath("/foo/bar/baz") |> System.IO.FileInfo).Directory` is more readable :)